server/paymentProviders/stripe/common.ts

```diff
@@ -181,3 +183,181 @@ export const createChargeTransactions = async (charge, { order }) => {
+  });
+
+  stripePaymentMethod = await stripe.paymentMethods.attach(stripePaymentMethod.id, {
+    customer: platformCustomer,
+  });
+
+  return await paymentMethod.update({
+    customerId: platformCustomer,
+    data: {
+      ...paymentMethod.data,
+      stripePaymentMethodId: stripePaymentMethod.id,
+      fingerprint: stripePaymentMethod.card.fingerprint,
+    },
+  });
+}
+
+export async function getOrCloneCardPaymentMethod(
+  platformPaymentMethod: typeof models.PaymentMethod,
+  collective: typeof models.Collective,
+  hostStripeAccount: string,
+  hostCustomer: string,
+) {
+  let platformCardTokenCardId = platformPaymentMethod.data?.stripePaymentMethodId;
+  let platformCardFingerprint = platformPaymentMethod.data?.fingerprint;
+
+  // store platform card payment method id and fingerprint for reuse.
+  if (!platformCardTokenCardId || !platformCardFingerprint) {
+    const platformCardToken = await stripe.tokens.retrieve(platformPaymentMethod.token);
+    platformCardFingerprint = platformCardToken.card.fingerprint;
+    platformCardTokenCardId = platformCardToken.card.id;
+
+    await platformPaymentMethod.update({
+      data: {
+        ...platformPaymentMethod?.data,
+        fingerprint: platformCardFingerprint,
+        stripePaymentMethodId: platformCardTokenCardId,
+      },
+    });
+  }
+
+  const cardPaymentMethodOnHostAccount = await models.PaymentMethod.findOne({
+    where: {
+      service: PAYMENT_METHOD_SERVICE.STRIPE,
+      type: PAYMENT_METHOD_TYPE.CREDITCARD,
+      CollectiveId: collective.id,
+      data: {
+        stripeAccount: hostStripeAccount,
+        fingerprint: platformCardFingerprint,
+      },
+    },
+  });
+
+  // card was already cloned to fiscal host customer account
+  if (cardPaymentMethodOnHostAccount) {
+    return cardPaymentMethodOnHostAccount;
+  }
+
+  // clone payment method to host stripe account
+  let clonedPaymentMethod = await stripe.paymentMethods.create(
+    {
+      // this is the customer on the platform account which holds the original card
+      customer: platformPaymentMethod.customerId,
+      // eslint-disable-next-line camelcase
+      payment_method: platformCardTokenCardId,
+    },
+    {
+      stripeAccount: hostStripeAccount,
+    },
+  );
+
+  // we attach the newly cloned payment method to the customer in the fiscal host account
+  clonedPaymentMethod = await stripe.paymentMethods.attach(
+    clonedPaymentMethod.id,
+    {
+      customer: hostCustomer,
+    },
+    {
+      stripeAccount: hostStripeAccount,
+    },
+  );
+
+  return await models.PaymentMethod.create({
+    customerId: hostCustomer,
+    CollectiveId: collective.id,
+    service: PAYMENT_METHOD_SERVICE.STRIPE,
+    type: PAYMENT_METHOD_TYPE.CREDITCARD,
+    confirmedAt: new Date(),
+    token: clonedPaymentMethod.id,
+    data: {
+      stripePaymentMethodId: clonedPaymentMethod.id,
+      stripeAccount: hostStripeAccount,
+      ...clonedPaymentMethod[clonedPaymentMethod.type],
+    },
+  });
```

Comment:
Yes, it would not be host-agnostic anymore, it would behave the same way as the SEPA/ACH do and maybe future payment method with the limitation to clone.

The alternative would be the store the payment method id in the data field of the platform card PM, which is more or less what we were doing. wdyt?

Reply:
I like the idea of being consistent with SEPA/ACH, but we need to make sure this isn't a bigger refactor than expected. To be honest, I'm really not sure of all side-effects that can come out of this refactor.

Things that come to mind:
- Possibly displaying multiple payment methods in the contribution flow that is in fact, the same card;
- `customerId` has been used to store the Platform customer id, are we sure we're covering the whole logic?
- What happens with existing payment methods, do we need to migrate them?

Reply:
I will experiment with another change to see if the payment method creation can be avoided. I will keep this one on hold for a moment.

> Possibly displaying multiple payment methods in the contribution flow that is in fact, the same card;
The cloned payment methods don't have `saved` set, they are not returned by the payment method list.

> customerId has been used to store the Platform customer id, are we sure we're covering the whole logic?
I believe the customerId in this case was only used to clone the payment method, this should have been refactored on this PR.

> What happens with existing payment methods, do we need to migrate them?
They wont be migrated. When they are used we will continue to clone them, but now it will be to a single host customer.

Reply:
@kewitz I refactored it to use only the platform payment method